synapse/logging/opentracing.py

```diff
@@ -909,8 +909,9 @@ def tag_args(func: Callable[P, R]) -> Callable[P, R]:
     @wraps(func)
     def _tag_args_inner(*args: P.args, **kwargs: P.kwargs) -> R:
         argspec = inspect.getfullargspec(func)
-        for i, arg in enumerate(argspec.args[1:]):
-            set_tag("ARG_" + arg, str(args[i]))  # type: ignore[index]
+        # We use `[1:]` to skip the `self` object reference
+        for i, arg in enumerate(args[1:]):
+            set_tag("ARG_" + argspec.args[i + 1], str(arg))  # type: ignore[index]
         set_tag("args", str(args[len(argspec.args) :]))  # type: ignore[index]
         set_tag("kwargs", str(kwargs))
         return func(*args, **kwargs)
```

Comment:
> use argument names in the tags(`ARG_limit` rather than `ARG_3`)

It already uses the arg names btw

---

From the edit, seems like this isn't viable or better than current so opting to leave it ⏩

Reply:
Argh, so it does. Sorry!

Reply:
No worries, it would be cool to just use this on whatever function and have it figure out if it needs to ignore `self`

Reply:
To make this handle any func, would it be better if we just ignore the first arg if it's named `self` or `cls`? It's not fool-proof but handles the idiomatic cases

Reply:
Sounds like the best compromise. I'd say make another PR for it if turns out we need it?

Reply:
I've added a `FIXME` comment to document the thoughts ⏩